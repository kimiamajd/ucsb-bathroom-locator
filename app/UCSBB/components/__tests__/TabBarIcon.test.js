import React from 'react';
import TabBarIcon from '../TabBarIcon';
import TestRenderer from 'react-test-renderer';
import { Platform } from 'react-native';

describe ('Tab Bar Icon', () => {
    it('should render the unfocused Map icon', () => {
        const focused = false;
        const component = TestRenderer.create(<TabBarIcon focused={focused} name={Platform.OS === 'ios' ? `ios-map`: 'md-map'}/>).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the unfocused Bathrooms icon and match to snapshot', () => {
        const focused = false;
        const component = TestRenderer.create(<TabBarIcon focused={focused} name={Platform.OS === 'ios' ? `ios-list-box`: 'md-list-box'}/>).toJSON();        
        expect(component).toMatchSnapshot();
    });

    it('should render the unfocused Settings icon and match to snapshot', () => {
        const focused = false;
        const component = TestRenderer.create(<TabBarIcon focused={focused} name={Platform.OS === 'ios' ? `ios-options`: 'md-options'}/>).toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render the focused Map icon', () => {
        const focused = true;
        const component = TestRenderer.create(<TabBarIcon focused={focused} name={Platform.OS === 'ios' ? `ios-map`: 'md-map'}/>).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the focused Bathrooms icon and match to snapshot', () => {
        const focused = true;
        const component = TestRenderer.create(<TabBarIcon focused={focused} name={Platform.OS === 'ios' ? `ios-list-box`: 'md-list-box'}/>).toJSON();        
        expect(component).toMatchSnapshot();
    });

    it('should render the focused Settings icon and match to snapshot', () => {
        const focused = true;
        const component = TestRenderer.create(<TabBarIcon focused={focused} name={Platform.OS === 'ios' ? `ios-options`: 'md-options'}/>).toJSON();
        expect(component).toMatchSnapshot();
    });
});