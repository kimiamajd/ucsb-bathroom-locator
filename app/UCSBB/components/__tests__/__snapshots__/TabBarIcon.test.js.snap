// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Bar Icon should render the focused Bathrooms icon and match to snapshot 1`] = `<Text />`;

exports[`Tab Bar Icon should render the focused Map icon 1`] = `<Text />`;

exports[`Tab Bar Icon should render the focused Settings icon and match to snapshot 1`] = `<Text />`;

exports[`Tab Bar Icon should render the unfocused Bathrooms icon and match to snapshot 1`] = `<Text />`;

exports[`Tab Bar Icon should render the unfocused Map icon 1`] = `<Text />`;

exports[`Tab Bar Icon should render the unfocused Settings icon and match to snapshot 1`] = `<Text />`;
